src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -285,101 +307,126 @@ object RuleUtils {
-            // index data is not updated with those files. Therefore, we need to handle
-            // appendedFiles and deletedFiles in IndexLogEntry.
-            (index.deletedFiles, index.appendedFiles.map(f => new Path(f.name)).toSeq)
-          } else {
-            val curFiles = Hyperspace
-              .getContext(spark)
-              .sourceProviderManager
-              .allFiles(baseRelation)
-              .map(f => FileInfo(f, index.fileIdTracker.addFile(f), asFullPath = true))
-            if (HyperspaceConf.hybridScanDeleteEnabled(spark) && index.hasLineageColumn) {
-              val (exist, nonExist) = curFiles.partition(index.sourceFileInfoSet.contains)
-              val filesAppended = nonExist.map(f => new Path(f.name))
-              if (exist.length < index.sourceFileInfoSet.size) {
-                (index.sourceFileInfoSet -- exist, filesAppended)
-              } else {
-                (Nil, filesAppended)
-              }
-            } else {
-              // Append-only implementation of getting appended files for efficiency.
-              // It is guaranteed that there is no deleted files via the condition
-              // 'deletedCnt == 0 && commonCnt > 0' in isHybridScanCandidate function.
-              (
-                Nil,
-                curFiles.filterNot(index.sourceFileInfoSet.contains).map(f => new Path(f.name)))
-            }
-          }
-
-        val filesToRead = {
-          if (useBucketSpec || !index.hasParquetAsSourceFormat || filesDeleted.nonEmpty ||
-              location.partitionSchema.nonEmpty) {
-            // Since the index data is in "parquet" format, we cannot read source files
-            // in formats other than "parquet" using one FileScan node as the operator requires
-            // files in one homogenous format. To address this, we need to read the appended
-            // source files using another FileScan node injected into the plan and subsequently
-            // merge the data into the index data. Please refer below [[Union]] operation.
-            // In case there are both deleted and appended files, we cannot handle the appended
-            // files along with deleted files as source files do not have the lineage column which
-            // is required for excluding the index data from deleted files.
-            // If the source relation is partitioned, we cannot read the appended files with the
-            // index data as the schema of partitioned files are not equivalent to the index data.
-            unhandledAppendedFiles = filesAppended
-            index.content.files
-          } else {
-            // If BucketSpec of index data isn't used (e.g., in the case of FilterIndex currently)
-            // and the source format is parquet, we could read the appended files along
-            // with the index data.
-            index.content.files ++ filesAppended
-          }
```

Comment:
Extracted as the `getFilesToReadAndUnhandled` method at [L341-L362](https://github.com/microsoft/hyperspace/pull/321/files#diff-8c312528287b51f613809d58a7b14b23c151d65794c099b653605795e0a738edR341-R362)